main_location.py

```diff
@@ -0,0 +1,82 @@
+import csv
+from dataclasses import dataclass
+
+
+@dataclass
+class LocalWeather:
+    location: str
+    max_temperature: int
+    min_temperature: int
+    avg_temperature: float
+    avg_humidity: float
+
+    @classmethod
+    def weather_by_location(cls, location, temp, humid):
+        max_temp = max(temp)
+        min_temp = min(temp)
+        avg_temp = average(temp)
+        avg_humid = average(humid)
+
+        return cls(
+            location = location,
+            max_temperature = max_temp,
+            min_temperature = min_temp,
+            avg_temperature = avg_temp,
+            avg_humidity = avg_humid
+        )
+    
+    def print_metrics(self):
+        print(f"Location:{self.location}")
+        print(f"Maximum temperature: {self.max_temperature}")
+        print(f"Minimum temperature: {self.min_temperature}")
+        print(f"Average temperature: {self.avg_temperature}")
+        print(f"Avearge humidity: {self.avg_humidity}")
+
+
+def average(lst):
+    return sum(lst) / len(lst)
+
+
+def file_reader(filename):
+    with open(filename, "r") as csvfile:
+        reader = csv.reader(csvfile)
+        next(reader)
+
+        weather_data = {}
+
+        for row in reader:
+            location = row[3]
+            temp = float(row[1])
+            humid = float(row[2])
+
+            if location not in weather_data:
+                weather_data[location] = {'temp': [], 'humid': []}
+
+            weather_data[location]['temp'].append(temp)
+            weather_data[location]['humid'].append(humid)
+
+    return weather_data
+
+def analyze_weather_data(weather_data):
+    weather_metrics = []
+
+    for location, data in weather_data.items():
+        temp = data['temp']
+        humid = data['humid']
+
+        weather = LocalWeather.weather_by_location(location, temp,humid)
+        weather_metrics.append(weather)
+
+    return weather_metrics
+
+
+def print_weather_metrics(weather_metrics):
+    for weather in weather_metrics:
+        weather.print_metrics()
+        print()
```

Comment:
This can probably just be done from main, since the code is only executed once, and it's pretty clear what it's function is. Functions can be saved for when logic gets more complex and/or when we need to use the code more than once